Gems/Atom/Tools/MaterialEditor/Code/Source/Document/MaterialDocument.cpp

```diff
@@ -377,23 +367,18 @@ namespace MaterialEditor
 
         // create source data from properties
         MaterialSourceData sourceData;
-        sourceData.m_materialType = m_materialSourceData.m_materialType;
-        
-        AZ_Assert(m_materialAsset && m_materialAsset->GetMaterialTypeAsset(), "When IsOpen() is true, these assets should not be null.");
         sourceData.m_materialTypeVersion = m_materialAsset->GetMaterialTypeAsset()->GetVersion();
+        sourceData.m_materialType = AtomToolsFramework::GetExteralReferencePath(normalizedSavePath, m_materialSourceData.m_materialType);
```

Comment:
There is a case I was working on fixing, which I think is not covered by your changes here. https://github.com/o3de/o3de/pull/5143/files
But most of the code I was adding is redundant since your GetExteralReferencePath uses GetSourceInfoBySourcePath too. I think you can fix it by just doing this...
```suggestion
        sourceData.m_materialType = AtomToolsFramework::GetExteralReferencePath(normalizedSavePath, AssetUtils::ResolvePathReference(m_absolutePath, m_materialSourceData.m_materialType));
```

Actually, I think this needs to be done in SaveAsCopy too (though I missed that in my PR).

Reply:
I will add a comment but in the open function, I'm storing absolute paths for the parent material and material type.